Release 1.2.0

turtlemoji opened the pull request:
* fix: integration runs
  - Fixed runs failing because integration / token problems

* feat: ignore file
  - Added the ability to ignore files or hex strings on the check

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced support for a `.checkcryptoignore` file, allowing users to ignore specific files, directories, or known false positive hex strings during crypto key detection scans.
  * Added detailed guidance in scan reports on how to use the ignore file to filter out false positives.

* **Documentation**
  * Updated README with instructions and examples for configuring ignore rules and specifying required GitHub permissions.
  * Added an example `.checkcryptoignore` configuration file.

* **Bug Fixes**
  * Improved error handling for notification steps to prevent workflow failures if check runs or comments cannot be created.

* **Tests**
  * Added tests to verify ignore file parsing and ignore logic.

* **Chores**
  * Updated workflow files to enhance permissions management and support multiple languages in CodeQL analysis.
  * Bumped version to 1.2.0.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

turtlemoji committed:
fix: integration runs (#190)

* fix: integration runs

* feat: ignore file

* fix: doc

* test: ignore

* test: ignore

* test: ignore file

* test: path

* fix: ignore path

* test: folder

* test: debug

* fix: comment

* test: folder ignore

* chore: add ignored pk

turtlemoji committed:
feat: false positive recommendation (#192)

* feat: false positive recommendation

* test: debug leak

* test: public key

* fix: action test

* fix: test action

* fix: test

* fix: action input

* fix: summary

* test: delete debug

turtlemoji committed:
fix: code ql (#193)

* fix: codeql advanced

turtlemoji committed:
docs: permissions (#194)

* docs: permissions

* test: leak

* test: deleted deleteme

* test: debug

* fix: comment

* test: remove debug

turtlemoji committed:
fix: unneeded permission (#195)

* test: permissions

* test: remove debug

jahabeebs commented:
@coderabbitai full review

jahabeebs commented on __tests__/ignore.test.ts:
```diff
@@ -0,0 +1,127 @@
+    expect(rules.stringIgnores.size).toBe(0);
+  });
+
+  test('should parse file patterns', () => {
```
nit: usually we avoid writing "should" in all the test descriptions as it gets quite repetitive 😉

turtlemoji replied:
fixed https://github.com/defi-wonderland/check-crypto-action/pull/191/commits/7dec0c2934628e68af7beb82266d2758a133e6bc

jahabeebs commented on action.yml:
```diff
@@ -1,6 +1,10 @@
 description: 'An action to check for crypto private keys in pushed code'
 author: 'Wonderland'
+# Required permissions:
+#   contents: read (always required - to read repository code and git diffs)
```
did you mean to delete this commented out code?

turtlemoji replied:
just a comment so users know which permissions they have to set when using it (if they want granularity)

turtlemoji committed:
fix: update package.json

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Turtle <78362532+turtlemoji@users.noreply.github.com>

turtlemoji committed:
chore: update dist (#197)

* chore: update dist

* fix: deterministic build

turtlemoji committed:
refactor: test descriptions (#198)

coderabbitai[bot] committed:
docs: add docstrings (#196)

* 📝 Add docstrings to `dev`

Docstrings generation was requested by @turtlemoji.

* https://github.com/defi-wonderland/check-crypto-action/pull/191#issuecomment-3028927988

The following files were modified:

* `src/ignore.ts`
* `src/main.ts`

* chore: update dist

* chore: update dist

* chore: update dist again?

* fix: dist

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Co-authored-by: Turtle <78362532+turtlemoji@users.noreply.github.com>
Co-authored-by: turtlemoji <turtlemoji@protonmail.com>

turtlemoji committed:
feat: glob patterns (#200)

* feat: glob patterns

* docs: update docs
